controllers/flowlogspipeline/flp_objects.go

```diff
@@ -255,10 +258,69 @@ func (b *builder) addTransformStages(lastStage *config.PipelineBuilderStage) {
 	agg := enrichedStage.Aggregate("aggregate", []api.AggregateDefinition{})
 	agg.EncodePrometheus("prometheus", api.PromEncode{
 		Port:   int(b.desired.PrometheusPort),
-		Prefix: "flp_",
+		Prefix: "netobserv_",
 	})
 }
 
+func (b *builder) mergeMetricsConfiguration(pipeline config.PipelineBuilderStage) {
+	// read in the metrics configuration file
+	yFileBytes, err := configflp.FlpMetricsConfig.ReadFile("flowlogs-pipeline-metrics.conf.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Merge Metrics Configuration: \n%s\n", string(yFileBytes))
+	var metricsConfig struct {
+		Pipeline   []config.Stage
+		Parameters []config.StageParam
+	}
+	err2 := yaml.Unmarshal(yFileBytes, &metricsConfig)
+	if err2 != nil {
+		log.Fatal(err2)
+	}
+
+	// merge the new items into the existing parameters configuration
+	parameters := pipeline.GetStageParams()
+
+	// obtain pointers to various parameters structures:
+	var tn1, tn2 *api.TransformNetwork
+	var ea1, ea2 []api.AggregateDefinition
+	var pe1, pe2 *api.PromEncode
+	var extractIndex int
+
+	for i, p := range parameters {
+		if p.Transform != nil && p.Transform.Network != nil {
+			tn1 = p.Transform.Network
+		}
+		if p.Extract != nil && p.Extract.Aggregates != nil {
+			ea1 = p.Extract.Aggregates
+			extractIndex = i
+		}
+		if p.Encode != nil && p.Encode.Prom != nil {
+			pe1 = p.Encode.Prom
+		}
+	}
+	for _, p := range metricsConfig.Parameters {
+		if p.Transform != nil && p.Transform.Network != nil {
+			tn2 = p.Transform.Network
+		}
+		if p.Extract != nil && p.Extract.Aggregates != nil {
+			ea2 = p.Extract.Aggregates
+		}
+		if p.Encode != nil && p.Encode.Prom != nil {
+			pe2 = p.Encode.Prom
+		}
+	}
+	// combine the rules in both lists
+	rules := append(tn1.Rules, tn2.Rules...)
+	tn1.Rules = rules
+
+	aggs := append(ea1, ea2...)
+	parameters[extractIndex].Extract.Aggregates = aggs
+
+	promMetrics := append(pe1.Metrics, pe2.Metrics...)
+	pe1.Metrics = promMetrics
```

Comment:
Same here, we need to check that pe1 is not nil

Reply:
pe1 is built upon the fact that there already exists a Encode.Prom in addTransformStages.